src/extensions/lapack/lapack-generics.lisp

```diff
@@ -2,23 +2,34 @@
 ;;;;
 ;;;; Author: Cole Scott
 
-(in-package #:magicl)
+(in-package #:magicl-lapack)
 
-(defgeneric mult (a b &key target alpha beta transa transb)
-  (:documentation "Multiply a by b, storing in target or creating a new tensor if target is not specified.
-Target cannot be the same as a or b."))
+;;; We use the convention "FOO-EXTENSION" to refer to this extension's
+;;; generic function implementation of MAGICL:FOO.
 
-(defgeneric lapack-eig (matrix))
+(magicl:define-extensible-function (magicl:mult mult-extension :lapack) (a b &key target alpha beta transa transb))
 
-(defgeneric lapack-hermitian-eig (matrix))
+(magicl:define-extensible-function (magicl:eig lapack-eig :lapack) (matrix))
 
-(defgeneric lapack-lu (matrix))
+(magicl:define-extensible-function (magicl:hermitian-eig lapack-hermitian-eig :lapack) (matrix))
 
-(defgeneric lapack-inv (matrix))
+(magicl:define-extensible-function (magicl:lu lapack-lu :lapack) (matrix))
+
+(magicl:define-extensible-function (magicl:inv lapack-inv :lapack) (matrix))
+
+(magicl:define-extensible-function (magicl:csd csd-extension :lapack) (matrix p q))
 
 (defgeneric lapack-csd (matrix p q))
```

Comment:
Should this also be changed to `magicl:define-extensible-function`

Reply:
No, the reason is because `lapack-csd` doesn't actually implement `magicl:csd`. The call chain is `magicl:csd -> csd-extension -> lapack-csd -> ZUNCSD`. I don't see why we need so much ceremony, but that was there already before this PR.